apps/viewer/src/app.tsx

```diff
@@ -72,17 +51,48 @@ function App() {
   }, []);
 
   useEffect(() => {
-    const newTrackSourcesStates = new Map(trackSourcesStates);
-    remoteTrackSources.forEach((source) => {
-      if (!newTrackSourcesStates.get(source.sourceId)) newTrackSourcesStates.delete(source.sourceId);
-    });
-    if (newTrackSourcesStates.size !== trackSourcesStates.size) {
-      setTrackSourcesStates(newTrackSourcesStates);
+    if (remoteTrackSources.size && !hasInitialisedMainStreamRef.current) {
+      const [[firstSourceId]] = remoteTrackSources;
+
+      setMainSourceId(firstSourceId);
+
+      projectToMainStream(firstSourceId);
+
+      hasInitialisedMainStreamRef.current = true;
```

Comment:
RemoteTrackSource is not empty might doesn't mean we have got the `mainStream` from the track event, isn't it? I suggest we check the `mainStream` state from `useViewer` hook directly.